packages/expect/src/__tests__/matchers.test.js

```diff
@@ -2240,3 +2321,66 @@ describe('toMatchObject()', () => {
     ).toThrowErrorMatchingSnapshot();
   });
 });
+
+describe('.toHaveBeenCalledWith', () => {
+  [
+    [
+      [1, 2, 3],
+      [1, 2, 3],
+    ],
+    [
+      [1, 2, 3],
+      [jestExpect.any(Number), jestExpect.any(Number), jestExpect.any(Number)],
+    ],
+    // issue 12463
+    [
+      [1, () => {}],
+      [1, jestExpect.optionalFn()],
+    ],
+    [
+      [1, undefined],
+      [1, jestExpect.optionalFn()],
+    ],
+    [[1], [1, jestExpect.optionalFn()]],
+  ].forEach(([received, expected]) => {
+    test(`{pass: true} expect(f).toHaveBeenCalledWith(...${stringify(
+      expected,
+    )}) when called with ${stringify(received)}`, () => {
+      const f = jest.fn();
+      f(...received);
+      jestExpect(f).toHaveBeenCalledWith(...expected);
+    });
+  });
+  [
+    [
+      [1, 2],
+      [1, 2, 3],
+    ],
+    [
+      [1, 2, 3],
+      [1, 2],
+    ],
+    [
+      [1, 2, 3],
+      [1, 2, 4],
+    ],
```

Comment:
This one is similar to https://github.com/facebook/jest/blob/506a0629147b2cad1469709b0b3e40bcaddc1a8d/packages/expect/src/__tests__/spyMatchers.test.ts#L217